src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,14 +27,26 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
+		value = address.trim();
+		String[] addressParts = splitAddress();
+		
+        if (!isValidAddress(address) || addressParts.length != EXPECTED_SIZE_OF_ADDRESS_ARRAY) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+
+        block = new Block(Integer.parseInt(addressParts[0]));
+        street = new Street(addressParts[1]);
+        unit = new Unit(addressParts[2]);
+        postalCode = new PostalCode(addressParts[3]);
     }
 
+	private String[] splitAddress() {
+		String[] addressParts = value.split(", ");
```

Comment:
you may want to re think how to handle spaces in the address. this split doesn't handle address like: 123,clementi ave3,#09-23,S112345